sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py

```diff
@@ -1761,9 +1761,13 @@ def get_blob_client(
             transport=TransportWrapper(self._pipeline._transport), # pylint: disable = protected-access
             policies=self._pipeline._impl_policies # pylint: disable = protected-access
         )
+        version_id = None
+        if isinstance(blob, BlobProperties):
+            version_id = blob['version_id']
```

Comment:
Could you change this to use a try/catch? I think that's the more common style. And could you move it to line 1760?

```py
try:
    version_id = blob['version_id']
except AttributeError:
    version_id = None
```

Reply:
Since this is Python, I don't _believe_ you need to pre-decalre `version_id` since its guaranteed to be set either in try or except.